lib/layout/list_view.dart

```diff
@@ -106,22 +105,43 @@ class ListViewWidgetState extends WidgetState<ListViewWidget> with TemplatedWidg
   Widget _buildBoxWidget(List<Widget> children) {
     // propagate text styling to all its children
     return DefaultTextStyle.merge(
-        style: TextStyle(
-            fontFamily: widget._controller.fontFamily,
-            fontSize: widget._controller.fontSize != null
-                ? widget._controller.fontSize!.toDouble()
-                : null),
-        child: SizedBox(
-          width: widget._controller.width != null ? widget._controller.width!.toDouble() : null,
-          height: widget._controller.height != null ? widget._controller.height!.toDouble() : null,
-          child: flutter.ListView.builder(
-              scrollDirection: Axis.vertical,
-              physics: const ScrollPhysics(),
-              itemCount: children.length,
-              shrinkWrap: true,
-              itemBuilder: (BuildContext context, int index) {
-                return children[index];
-              }),
-        ),);
+      style: TextStyle(
+          fontFamily: widget._controller.fontFamily,
+          fontSize: widget._controller.fontSize != null
+              ? widget._controller.fontSize!.toDouble()
+              : null),
+      child: Container(
+        width: widget._controller.width?.toDouble(),
+        height: widget._controller.height?.toDouble(),
+        decoration: _buildBoxDecoration(),
+        child: flutter.ListView.builder(
+            padding: widget._controller.padding ?? const EdgeInsets.all(0),
+            scrollDirection: Axis.vertical,
+            physics: const ScrollPhysics(),
+            itemCount: children.length,
+            shrinkWrap: true,
+            itemBuilder: (BuildContext context, int index) {
+              return GestureDetector(
+                  onTap: widget._controller.onItemTap == null ? null : () =>
+                      ScreenController().executeAction(context, widget._controller.onItemTap!),
+                  child: children[index]);
+            }),
+      ),);
+  }
+  BoxDecoration _buildBoxDecoration() {
```

Comment:
did you copy this from some other widgets? can you re-use it as function instead?